hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ExpressionIndexSupport.scala

```diff
@@ -78,8 +93,241 @@ class ExpressionIndexSupport(spark: SparkSession,
     }
   }
 
+  /**
+   * Loads view of the Column Stats Index in a transposed format where single row coalesces every columns'
+   * statistics for a single file, returning it as [[DataFrame]]
+   *
+   * Please check out scala-doc of the [[transpose]] method explaining this view in more details
+   */
+  def loadTransposed[T](targetColumns: Seq[String],
```

Comment:
Some of the code here is copied directly from `ColumnStatsIndexSupport` with modifications to handle binary expressions, and we need to refactor in a followup

Reply:
Created https://issues.apache.org/jira/browse/HUDI-8649